krnlmon_main.h

```diff
@@ -13,7 +13,7 @@
 extern "C" {
 #endif	
 int krnlmon_init(void);
```

Comment:
1. The header file is not dependent on any of the definitions in `pthread.h`, so it should not #include it. This is the responsibility of the [source] files that require pthread definitions. This principle is called "include what you use."
2. I recommend adding a blank lines before and after the function prototypes, separating them from the __cplusplus #ifdef blocks. It makes the file easier to read.

Reply:
Incidentally, the symbol to specify before including pthread.h should be `__USE_GNU`, not `_GNU_SOURCE`.

Reply:
Done